reconcile/ocm_update_recommended_version.py

```diff
@@ -0,0 +1,128 @@
+import functools
+from typing import Any
+
+import semver
+from reconcile import queries, mr_client_gateway
+from reconcile.ocm.types import OCMSpec
+from reconcile.utils.mr.ocm_update_recommended_version import (
+    CreateOCMUpdateRecommendedVersion,
+    UpdateInfo,
+    WorkloadRecommendedVersion,
+)
+from reconcile.utils.ocm import OCMMap
+
+QONTRACT_INTEGRATION = "ocm-update-recommended-version"
+
+
+def get_highest(version_set: set[str]) -> str:
+    sorted_version_set = sorted(
+        version_set, key=functools.cmp_to_key(semver.compare), reverse=True
+    )
+    return sorted_version_set[0]
+
+
+def get_majority(version_set: set[str], versions: list[str]) -> str:
+    return max(version_set, key=versions.count)
+
+
+def recommended_version(
+    versions: list[str], high_weight: int = 0, majority_weight: int = 0
```

Comment:
Could you accept `versions` as an `Iterable` to keep this more general?

Reply:
This one might have been missed unless I'm looking at stale data? It might be more of a nit here, but the more that we tightly couple these types everywhere, the harder it is to reuse code, so it'd be good to keep in mind for other cases.